src/search/model/SearchModel.ts

```diff
@@ -118,6 +109,20 @@ export class SearchModel {
 				query,
 			}
 
+			const monitorHandle = progressTracker.registerMonitorSync(daysInMonths.length)
+			const monitor: IProgressMonitor = assertNotNull(progressTracker.getMonitor(monitorHandle))
+
+			if (this.cancelSignal()) {
```

Comment:
we haven't done anything async until this point, why is it checking for cancel here?

Reply:
We have the calendar model/repository that might be not initialized yet